view/base/web/js/model/address_validation_core.js

```diff
@@ -93,8 +89,46 @@ function (ko, $) {
             }
         },
 
+        isSuggestedAddress: function (address) {
+            var self = this;
+            var isSuggested = false;
+            var suggestedAddresses = this.suggestedAddresses();
+            suggestedAddresses.forEach(function(suggestedAddress, index) {
+                if (self.addressMatches(self.formatAddress(address), self.formatSuggestedAddress(suggestedAddress.address))) {
+                    isSuggested = true;
+                }
+            });
+            return isSuggested;
+        },
+
         updateSuggestedAddresses: function (addr) {
             this.suggestedAddresses(addr);
+        },
+
+        formatAddress: function (address) {
+            var formattedAddress = {
+                'street0': address.street[0],
+                'city': address.city,
+                'region': address.region_id,
+                'country': address.country_id,
+                'postcode': address.postcode
+            };
+            return formattedAddress;
```

Comment:
With this and `formatSuggestedAddress`, could you return the object instead of instantiating a var?

Reply:
Agreed, will get this updated.